services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_specs.py

```diff
@@ -182,6 +183,17 @@ def _update_resource_limits_and_reservations(
         spec["environment"] = environment
 
 
+def _update_container_labels(service_spec: ComposeSpecLabel, user_id: UserID) -> None:
+    for spec in service_spec["services"].values():
+        labels: set[str] = set(spec.get("labels", []))
+
+        # this labels is primarily used for metrics scraping
+
+        labels.add(get_prefixed_container_label("user.id", user_id))
```

Comment:
Some comments:
- I'm ok-ish to using the if statement to check.
- in the case where "labels" does not exist, it will return a new instance, so reassignment is mandatory

Reply:
for the second point I disagree. If you use ``setdefault``, as I write above, that does not happen

Reply:
I've read it wrong, sorry, missed the set default.